CAP/gap/CategoriesCategory.gi

```diff
@@ -426,6 +426,8 @@ InstallGlobalFunction( ApplyFunctor,
                 if not IsIdenticalObj( CapCategory( computed_value ), range_category ) then
                     Error( Concatenation( "the category of the result of the object function of the functor \"", Name(functor), "\" does not coincide with the range of this functor" ) );
                 fi;
+            else
+                Error( Concatenation("The result of the object function of the functor \"", Name(functor), "\" does not have a CAP category" ) );
```

Comment:
If I have understood the concept correctly, the case that `computed_value` has no CAP category is supported explicitly, and that is why `Add` is called later. I think we should test for `IsCapCategoryObject( computed_value ) = true` here.

Please correct me if I'm wrong.

Reply:
Yes, Fabian is right.
Please test for the filter `IsCapCategoryObject`
and analogously for the filter `IsCapCategoryMorphism` below.